src/components/TransactionDialog.vue

```diff
@@ -0,0 +1,165 @@
+<script lang="ts" setup>
+import { computed } from 'vue';
+import { BigNumber } from 'ethers/lib/ethers';
+import { useI18n } from 'vue-i18n';
+
+import { truncateAddress } from 'src/antelope/wallets/utils/text-utils';
+import { WEI_PRECISION, ZERO_ADDRESSES, formatWei } from 'src/lib/utils';
+
+import OutlineButton from 'src/components/OutlineButton.vue';
+
+const $i18n = useI18n();
+const $t = $i18n.t;
+const locale = $i18n.locale.value;
+
+const props = defineProps({
+    trx: {
+        type: Object,
+        default: null,
+    },
+});
+
+const actionText = computed(() => {
+    if (
+        !props.trx.parsedTransaction
+        && props.trx.from === ZERO_ADDRESSES
+        && props.trx.value
+        && parseInt(props.trx.gasPrice as string) === 0
+    ) {
+        return $t('pages.transactions.deposit_native');
+    }
+    if (
+        !props.trx.parsedTransaction
+        && props.trx.to === ZERO_ADDRESSES
+        && props.trx.value
+        && parseInt(props.trx.gasPrice as string) === 0
+    ) {
+        return $t('pages.transactions.withdraw_native');
+    }
+
+    if (!props.trx.parsedTransaction && props.trx.to === null && props.trx.data !== null) {
+        return $t('pages.transactions.contract_deployment');
+    }
+
+    if (props.trx.parsedTransaction) {
+        return props.trx.parsedTransaction.name;
+    }
+
+    if (!props.trx.parsedTransaction && props.trx.input === '0x' && props.trx.value) {
+        // action is a transfer; handle separately in the template
+        return '';
+    }
+
+    return props.trx.input.slice(0, 10);
+});
+const gasLimit = computed(() => {
+    const gas = BigNumber.from(props.trx.gasLimit);
+
+    try {
+        return gas.toNumber().toLocaleString(locale);
+    } catch (e) {
+        console.error(e);
+        return gas.toString();
+    }
+});
+const gasUsed = computed(() => {
+    const gas = BigNumber.from(props.trx.gasUsed ?? props.trx.gasused);
```

Comment:
Why is there this fallback? which source returns `gasused `instead `gasUsed`?

Reply:
I don't know honestly, it was in the code when I took over. i'll check if it is needed

Reply:
yeah so the address/transactions endpoint has `gasused` and the normal transactions endpoint has `gasUsed`